File: Leetcode/Easy/Python/70 - Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        uniqueWays = 0
        for x in range(n+1):
            for y in range(n):
                if x + 2*y == n:
                    uniqueWays += 1
        return uniqueWays

solution = Solution()
print(solution.climbStairs(1))
print(solution.climbStairs(2))
print(solution.climbStairs(3))
print(solution.climbStairs(4))

File: Leetcode/Easy/JavaScript/_70 - Climbing Stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    var uniqueWays = 0;
    for(let i=0;i<n;i++){
        for(let j=0;j<=n;j++){
            if(i+2*j == n){
                uniqueWays++;
            }
        }
    }
    return uniqueWays;
};

console.log(climbStairs(0));
console.log(climbStairs(1));
console.log(climbStairs(2));
console.log(climbStairs(3));
console.log(climbStairs(4));
